hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -1137,45 +351,21 @@ public void stopBalancer()
       TimeoutException {
     lock.lock();
     try {
-      // should be leader, out of safe mode, and currently running
       validateState(true);
-      saveConfiguration(config, false, 0);
```

Comment:
We would need this change?

Reply:
saveConfiguration could end up happening outside lock if executed in the ContainerBalancerTask.

Reply:
For HA, SCM stop should not saveConfiguration to false, and CLI based stop, it should set to false for normal flow.